### PR TITLE
Fix qs using concatenated value|key

### PR DESCRIPTION
**Pull request by Fletch153**

### CODE REVIEW

File: .gas-snapshot [hunk 1]
```diff
@@ -469,18 +469,22 @@ ProductIdUtilTest:testFormatAndParseAreMirrored(uint8,uint8,uint8,uint8,uint8) (
...
-ArrayUtilTest:testArgSort() (gas: 8966)
-ArrayUtilTest:testArgSortDups() (gas: 8919)
-ArrayUtilTest:testArgSortDupsEvenItems() (gas: 11028)
-ArrayUtilTest:testArgSortDupsInt() (gas: 8963)
-ArrayUtilTest:testConcat() (gas: 5152)
+ArrayUtilTest:testArgSort() (gas: 13937)
+ArrayUtilTest:testArgSortDups() (gas: 13504)
+ArrayUtilTest:testArgSortDupsEvenItems() (gas: 16387)
+ArrayUtilTest:testArgSortDupsInt() (gas: 15055)
+ArrayUtilTest:testConcat() (gas: 5174)
```
Comment (antoncoding) [inline .gas-snapshot:476]: @dsshap given that the gas seems to be +40% for these methods, what were the intention behind this change?

File: src/libraries/ArrayUtil.sol [hunk 2]
```diff
@@ -698,4 +718,49 @@ library ArrayUtil {
...
         }
     }
+
+    /**
+     * @notice generates a key by concatenating two indexes price | index^decimals
+     * @param value the value to store
+     * @param key some index. The value of this key must never exceed the decimals
+     */
+    function valueToKey(uint256 value, uint256 index, uint256 decimals) internal pure returns (uint256 key) {
+        return value * 10 ** decimals + index;
```
[the remark this answers is not in the document]
Comment (dsshap) [reply src/libraries/ArrayUtil.sol:714]: @antoncoding

[the remark this answers is not in the document]
Comment (dsshap) [reply src/libraries/ArrayUtil.sol:714]: @antoncoding the indexes mixes up order of indexes for equal elements. For the array [1,1,1,1] the expected indexArray should be [0,1,2,3], but instead it is [2,3,0,1]

File: .gas-snapshot (hunk 1 above)
Comment (dsshap) [reply .gas-snapshot:476]: @antoncoding see above reply

**Comment by antoncoding** [comment]
Honestly, can't think of a good solution to this problem. I think adding this is terrible in terms of complexity and gas cost.

I'd definitely prefer implementing a different algorithm that gives you the "stability". 
https://en.wikipedia.org/wiki/Sorting_algorithm#Stability

Bubble sort is the easiest with stability, not the most efficient. merge sort seems reasonable

**Pull request closed without merging**